src/index.js

```diff
@@ -165,5 +180,71 @@ export default class ScrollPlugin extends Plugin {
                 swup.scrollTo(0, this.shouldAnimate('betweenPages'));
             }
         }
-    };
+    }
+
+    onWillReplaceContent = () => {
+        this.storeScrollPositions(this.previousUrl);
+        this.previousUrl = window.location.href;
+    }
+
+    onClickLink = (e) => {
+        this.deleteStoredScrollPositions(e.delegateTarget.href);
+    }
+
+    /**
+     * Stores the scroll positions for the current URL
+     * @param {string} url 
+     */
+    storeScrollPositions(url) {
+      // create an object to store the scroll positions
+      const storeEntry = {
+        window: {top: window.scrollY, left: window.scrollX},
+        elements: []
+      }
+      // fill up the object with scroll positions for each matching element
+      const $elements = document.querySelectorAll('[data-swup-restore-scroll]');
```

Comment:
Can we make the selector configurable? We could also exit early here if a normal `query` returns `null` and skip the `forEach`.

Reply:
Yes, I'll add the option for the selector. Right now, this really is just a POC to see what you think. Could you take it for a spin in some little test project? 

> We could also exit early here if a normal query returns null and skip the forEach.

`querySelectorAll` always returns a list, so no need to detect `null` and bail early here.

Reply:
Added [scrollContainersSelector](https://github.com/swup/scroll-plugin/pull/26/commits/df4ed5d054a7e2af9c188eb23cb91c7ccde319d2) to the options.